sklearn/decomposition/_kernel_pca.py

```diff
@@ -210,6 +263,21 @@ def _fit_transform(self, K):
                                                 tol=self.tol,
                                                 maxiter=self.max_iter,
                                                 v0=v0)
+        elif eigen_solver == 'randomized':
+            # not using flip_sign=True for deterministic output, we do it later
+            U, S, V = randomized_svd(K, n_components=n_components,
+                                     n_iter=self.iterated_power,
+                                     flip_sign=False,
+                                     random_state=self.random_state)
+            self.alphas_ = U[:, :n_components]  # eigenvectors
+            self.lambdas_ = S[:n_components]    # eigenvalues
+
+            # SVD does not guarantee that sign of u and v is the same. If they
+            # are different that means that the corresponding eigenvalue has
+            # the wrong sign, we have to fix it.
+            VU = np.dot(V[:n_components, :], U[:, :n_components])
+            signs = np.sign(np.diag(VU))
+            self.lambdas_ = self.lambdas_ * signs
```

Comment:
is this what `utils.extmath.svd_flip` does?

Reply:
No, `svd_flip` is an optional operation to return deterministic eigenvectors, while this is a mandatory operation to return the actual eigenvalues from singular values. I'll modify the comment in the code so that it is very well explained.